sklearn/tests/test_multioutput.py

```diff
@@ -561,3 +561,30 @@ class A(MultiOutputEstimator, MultiOutputRegressor):
 
     with pytest.warns(FutureWarning, match="is deprecated in version 0.22"):
         A(SGDRegressor(random_state=0, max_iter=5))
+
+
+class GradientBoostingRegressorTestFitParam(GradientBoostingRegressor):
+    def fit(self, X, y, sample_weight=None, monitor=None, **fit_params):
+        self.fit_params = fit_params
+        super().fit(X, y, sample_weight=sample_weight)
+
+
+def test_test_multi_target_regression_with_fit_params():
+    X, y = datasets.make_regression(n_targets=3)
+    X_train, y_train = X[:50], y[:50]
+    X_test, y_test = X[50:], y[50:]
+    sample_weight = np.random.random((50, ))
+
+    references = np.zeros_like(y_test)
+    for n in range(3):
+        rgr = GradientBoostingRegressorTestFitParam(random_state=0)
+        rgr.fit(X_train, y_train[:, n], sample_weight=sample_weight)
+        references[:, n] = rgr.predict(X_test)
+
+    rgr = GradientBoostingRegressorTestFitParam(random_state=0)
+    rgr = MultiOutputRegressor(rgr)
+    rgr.fit(X_train, y_train, sample_weight=sample_weight, should_succeed=True)
```

Comment:
Support for scalar value is an abuse of the scikit-learn API and we might deprecate it.
In our test suite, it would be best to have `fit_params` being arrays of the same size as `X_train`.